### PR TITLE
Update extension English docs

### PR DESCRIPTION
**Pull request by hjianbo**

### CODE REVIEW

File: en_US/advanced/lang-exproto.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+## Design
+
+![Extension-Protocol Arch](./assets/exproto-arch.jpg)
+
+The interfaces provided by emqx-exproto:
+
+1. **Connection Layer:** This layer mainly **maintains the life cycle of the socket, and the sending/receiving of data**. Included:
+    - Listens on a port. When a new TCP/UDP connection arrives, a connection process is started to maintain the state of the connection.
+    - Call the `OnSocketCreated` callback. Used to notify the user's server that **a new connection has been established**.
+    - Call `OnScoektClosed` callback. Used to notify the user's server that a connection **has been closed**.
```
Comment (k32) [inline en_US/advanced/lang-exproto.md]:
```suggestion
    - Call `OnSocketClosed` callback. Used to notify the user's server that a connection **has been closed**.
```

File: en_US/advanced/lang-exhook.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+![chain_of_responsiblity](./assets/chain_of_responsiblity.png)
+
+
+## APIs
+
+As the event handler, i.e. user implemented server side of gRPC. It can define the list of hooks that need to be mounted, and the implement callback functions for how to go about handling each event when it arrives.
+
+These interfaces are defined as a gRPC service called `HookProvider`:
+
+```protobuff
```
Comment (k32) [inline en_US/advanced/lang-exhook.md]:
```suggestion
```protobuf
```

**Comment by sigming** [comment]
In file `/en_US/advanced/hooks.md`, change `multiple-language-support.md` to `lang.md`

**Review by Rory-Z — APPROVED** [review]

**Review by sigming — APPROVED** [review]

File: en_US/advanced/lang-exproto.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+**Extension Protocol** is provided by the **emqx-exproto** plugin, which allows other programming languages (e.g. Python, Java, etc.) to process bytes directly for parsing private protocols and provides a Pub/Sub interface for message exchange with EMQ X.
+
+This feature gives EMQ X the power of scalability to handle any private protocol in a user-friendly programming language and enjoy the benefits of extremely high concurrent connections brought by the EMQ X.
+
+
+## Features
+
+- Extremely scalable. Supports all major programming languages using gRPC as the RPC communication framework
+- Fully asynchronous IO. the connection layer is implemented as fully asynchronous non-blocking I/O
```
Comment (HAHAZHUZHU) [inline en_US/advanced/lang-exproto.md]: the connection layer -> The connection layer

File: en_US/advanced/lang-exproto.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    - Provides the `Close` interface. Called by user's server. **For actively closing the connection**.
+
+2. **Protocol/Session Layer:** This layer primarily **provides the PUB/SUB interface** for message interoperability with the EMQ X. Includes:
+    - Provides the `Authenticate` interface. Called by user's server to register clients into EMQ X.
+    - Provides the `StartTimer` interface. Called by user's server to start a timer such as a heartbeat for the connected process.
+    - Provides the `Publish` interface. Called by user's server to publish messages to the EMQ X.
+    - Provides the `Subscribe` interface. Called by user's server to subscribe to a topic to receive certain downlink messages from the EMQ X.
+    - Provides the `Unsubscribe` interface. Called by user's server to unsubscribe from a topic.
+    - Calls the `OnTimerTimeout` callback. Used to handle timer timeout events
+    - Call the `OnReceivedMessages` callback. Used to receive downlink messages.
```
Comment (HAHAZHUZHU) [inline en_US/advanced/lang-exproto.md]: Used to receive downlink messages. -> Used to receive downlink messages (After a successful subscription to a topic, this method will be called back if there are messages on the topic)

File: en_US/advanced/lang-exproto.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+The user needs to implement the gRPC service of `ConnectionHandler` to receive callback events from EMQ X.
+
+The main development steps are as following:
+
+1. Copy the `lib/emqx_exproto-<x.y.z>/priv/protos/exproto.proto` file to your project.
+2. Generate the code for the gRPC server and client side of `exproto.proto` using the gRPC framework for the corresponding programming language.
+3. Implement the interfaces defined in exhook.proto on demand
+
+Once the development is complete, the service needs to be deployed to a server that can communicate with EMQ X.
```
Comment (HAHAZHUZHU) [inline en_US/advanced/lang-exproto.md]: that can communicate with EMQ X. -> that can communicate with EMQ X and ensure that the ports are open.

File: en_US/advanced/lang-exhook.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+---
+
+# Extension Hook
+
+The **Extension Hook** is supported by the **emqx-exhook** plugin. It allows users to process EMQ X's [Hooks](hooks.md) using other programming languages.
+
+In this way, other programming languages can handle emqx events for the purpose of customizing and extending emqx. For example, users can use other programming languages to implement:
+
+- Authorization for client access
+- ACL authentication for publish/subscribe
```
Comment (HAHAZHUZHU) [inline en_US/advanced/lang-exhook.md]: publish/subscribe -> publishing/subscribing

File: en_US/advanced/lang-exhook.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+- Persistence for messages
```
Comment (HAHAZHUZHU) [inline en_US/advanced/lang-exhook.md]: Persistence for messages -> Persistence and bridging for messages

File: en_US/advanced/lang-exhook.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+The user needs to implement the gRPC service of `HookProvider` to receive callback events from EMQ X.
+
+The main development steps are as following:
+
+1. Copy the `lib/emqx_exhook-<x.y.z>/priv/protos/exhook.proto` file to your project.
+2. Generate the code for the gRPC server side of `exhook.proto` using the gRPC framework for the corresponding programming language.
+3. Implement the interfaces defined in exhook.proto on demand
+
+Once the development is complete, the service needs to be deployed to a server that can communicate with EMQ X.
```
Comment (HAHAZHUZHU) [inline en_US/advanced/lang-exhook.md]: that can communicate with EMQ X. -> that can communicate with EMQ X and ensure that the ports are open.

**Comment by hjianbo** [comment]
@HAHAZHUZHU Fixed, thanks!

**Review by Rory-Z — APPROVED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**